pkg/volume/csi/csi_plugin.go

```diff
@@ -408,6 +408,8 @@ func (p *csiPlugin) NewMounter(
 	}
 	klog.V(4).Info(log("created path successfully [%s]", dataDir))
 
+	mounter.MetricsProvider = NewMetricsCsi(volumeHandle, dataDir)
```

Comment:
Should the path be dir or dataDir?  dir is the one we pass as the targetPath to NodePublish?

Reply:
good catch, this indeed should be `dir` and not `dataDir`.

Reply:
Can we make sure to validate this with a real CSI driver?

Reply:
Replaced the path. 
>Can we make sure to validate this with a real CSI driver?
>
Yes, thats the very next plan. I dont think, the CSI drivers exist which expose the metrics. I have started the work for Ceph-CSI any way.

Reply:
This can be also tested via CSI mock driver fwiw. Mock driver can implement `NodeGetVolumeStats` RPC call and it can check for volume path to be same as publish path and return error if it isn't.  

This has to go in an e2e though. @humblec can you add a github issue for creating an e2e for this?

Reply:
Sure @gnufied . I am opening an issue for e2e tracking as discussed earlier..

Reply:
@gnufied The E2E issue is here https://github.com/kubernetes/kubernetes/issues/77933 . Thanks !